### DISCUSSION TITLE
YALMIP modeling speed issues

### DISCUSSION THREAD

**Discussion by YANGseuJW** [yalmip/YALMIP#1443] [General discussion] [answered]
Hi, I have a question I need to ask. When I use yalmip to model the same MILP problem, the modeling speed slows down as the number of times I model increases, and I use the clc, clear commands before modeling, what is the reason for this? Whenever I invoke the gurobi solver, it is always the first modeling that is the fastest!

**Comment by johanlofberg** [comment, marked as answer]
Typical bad modelling is to repeatedly defining new sdpvars inside a loop, when you just as well can define them outside, and the construct new models inside loop. even further, sometimes there are onoly parts of a model that changes, and then one should simply define those and add inside the loop, and final improvement is to use optimizer if possible.

If you still have to redefine from scratch, you are probably missing a yalmip('clear')

**Reply by YANGseuJW** [reply]
Thank you, sir
